### PR TITLE
chore: provide more fine-grained websocket close code on errors

### PR DESCRIPTION
**Pull request by phantumcode**
*Issue #, if available:*

*Description of changes:*
* update web socket close to include liveness-specific close code
* update timeout logic start the timeout timer when the oval is initially displayed to handle edge case when a face is detected and the oval is displayed and the user moves away immediately

*Check points: (check or cross out if not relevant)*

- [x] Added new tests to cover change, if needed
- [x] Build succeeds with all target using Swift Package Manager
- [x] All unit tests pass
- [x] Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)
- [ ] ~Documentation update for the change if required~
- [x] PR title conforms to conventional commit style
- [ ] ~If breaking change, documentation/changelog update with migration instructions~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Sources/FaceLiveness/Views/Liveness/FaceLivenessDetectionView.swift
Comment (atierian) [inline Sources/FaceLiveness/Views/Liveness/FaceLivenessDetectionView.swift:252]: Looks good. One (non-blocking) suggestion - consider adding the websocket close frame code to the `LivenessStateMachine.LivenessError` struct rather than mapping in the switch statement.

```swift
struct LivenessError: Error, Equatable {
    let code: UInt8
    let webSocketCloseCode: URLSessionWebSocketTask.CloseCode?

    static let unknown = LivenessError(
        code: 0,
        webSocketCloseCode: .unexpectedRuntimeError
    )

    static let missingVideoPermission = LivenessError(
        code: 1,
        webSocketCloseCode: .missingVideoPermission
    )

    static let errorWithUnderlyingOSFramework = LivenessError(
        code: 2,
        webSocketCloseCode: .unexpectedRuntimeError
    )

    static let userCancelled = LivenessError(
        code: 3,
        webSocketCloseCode: .ovalFitUserClosedSession
    )

   // ...

    static func == (lhs: LivenessError, rhs: LivenessError) -> Bool {
        lhs.code == rhs.code
    }
}
```

Which will let you:
```swift
viewModel.livenessService.closeSocket(with: error.webSocketCloseCode ?? .normalClosure)
```

**Review by atierian — APPROVED** [review]
Recommend using a stored property for the `closeCode` rather than computed to avoid the switch and nil coelescing operators. Approving as is though, lmk if you want to make that change and need a new approval.

File: Sources/FaceLiveness/Views/Liveness/FaceLivenessDetectionView.swift
Comment (phantumcode) [reply Sources/FaceLiveness/Views/Liveness/FaceLivenessDetectionView.swift]: I've updated the code per suggestion, it does look a lot cleaner!

**Review by atierian — APPROVED** [review]

**Pull request merged**
